src/app/auth/guards/unsaved-changes.guard.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { CanDeactivate } from '@angular/router';
+
+export interface CanComponentLeave {
```

Comment:
Put this interface in separate interface file, preferably in an interface folder in /auth